ibis/expr/analysis.py

```diff
@@ -1154,7 +1122,8 @@ def _is_analytic(op):
 
 
 def is_reduction(expr):
```

Comment:
Memoization would spare quite some operation here (though would need benchmarks).

Reply:
Yeah, probably best to leave it as is for now until we get to a better state with `analysis.py`.